packages/client/src/client.js

```diff
@@ -31,13 +31,14 @@ function validateProjectPath(path) {
 export default class PercyClient {
   log = logger('client');
   env = new PercyEnvironment(process.env);
+  envWarning = false;
 
   constructor({
     // read or write token, defaults to PERCY_TOKEN environment variable
     token,
     // initial user agent info
-    clientInfo = '',
-    environmentInfo = '',
+    clientInfo = [],
+    environmentInfo = [],
```

Comment:
(I can't comment on the unchanged folded code below this, but I'll comment here because it's close enough)

Rather than initialize these properties in this constructor, we can move the `new Set()`s to the same place you added the `envWarning`. Then in the constructor, utilize the `this.add*Info()` methods to handle empty strings (which they already do).